test/testng/org/openjdk/jextract/test/json/parser/JSONParserTests.java

```diff
@@ -0,0 +1,622 @@
+        assertEquals(value.asArray().get(2).asInt(), 3);
+    }
+
+    @Test
+    void testParseObjectWithWhitespace() {
+        var value = parser.parse("   \t \r\n   {  \t \n \r\"a\" \n\n \t : \r\n 1 \n\n, \"b\"  :  2 ,  \"c\"  :  3  }  ");
+        assertEquals(value.asObject().get("a").asInt(), 1);
+        assertEquals(value.asObject().get("b").asInt(), 2);
+        assertEquals(value.asObject().get("c").asInt(), 3);
+    }
+
+    @Test
+    void testObjectShortcut() {
+        var value = parser.parse("{ \"a\":1, \"b\":2, \"c\":3 }");
+        assertEquals(value.get("a").asInt(), 1);
+        assertEquals(value.get("b").asInt(), 2);
+        assertEquals(value.get("c").asInt(), 3);
+    }
+
+    @Test
+    void testArrayShortcut() {
+        var value = parser.parse("[ 1, 2, 3]");
+        assertEquals(value.get(0).asInt(), 1);
+        assertEquals(value.get(1).asInt(), 2);
+        assertEquals(value.get(2).asInt(), 3);
+    }
+
+    @Test
+    void testIntToString() {
+        var v = JSON.of(17);
+        assertEquals(v.toString(), "17");
+    }
+
+    @Test
+    void testDoubleToString() {
+        var v = JSON.of(17.7);
+        assertEquals(v.toString(), "17.7");
+    }
+
+    @Test
+    void testBooleanToString() {
+        var v = JSON.of(true);
+        assertEquals(v.toString(), "true");
+    }
+
+    @Test
+    void testNullToString() {
+        var v = JSON.of();
+        assertEquals(v.toString(), "null");
+    }
+
+    @Test
+    void testStringToString() {
+        var v = JSON.of("Hello, JSON");
+        assertEquals(v.toString(), "\"Hello, JSON\"");
+    }
+
+    @Test
+    void testArrayToString() {
+        var v = new JSONArray();
+        v.add(1);
+        v.add(2);
+        v.add(3);
+        assertEquals(v.toString(), "[1,2,3]");
+    }
+
+    @Test
+    void testObjectToString() {
+        var v = new JSONObject();
+        v.put("a", 1);
+        v.put("b", 2);
+        v.put("c", 3);
+        assertEquals(v.toString(), "{\"a\":1,\"b\":2,\"c\":3}");
+    }
+
+    @Test
+    void testNestedObjectToString() {
+        var inner = new JSONObject();
+        inner.put("a", 1);
+        inner.put("b", 2);
+        inner.put("c", 3);
+
+        var outer = new JSONObject();
+        outer.put("inner", inner);
+        assertEquals(outer.toString(), "{\"inner\":{\"a\":1,\"b\":2,\"c\":3}}");
+    }
+
+    @Test
+    void testToStringAndParse() {
+        var inner = new JSONObject();
+        inner.put("a", 1);
+        inner.put("b", 2);
+        inner.put("c", 3);
+
+        var outer = new JSONObject();
+        outer.put("inner", inner);
+
+        var s = outer.toString();
+
+        var parsed = parser.parse(s);
+        assertEquals(parsed.get("inner").get("a").asInt(), 1);
+        assertEquals(parsed.get("inner").get("b").asInt(), 2);
+        assertEquals(parsed.get("inner").get("c").asInt(), 3);
+    }
+
+    @Test
+    void testLargerJSONText() {
+        var text = "{\n" +
```

Comment:
Would it make sense, for this or other big chunks of JSON, to do a roundtrip test where we parse the text, we get a JSONValue, then we obtain the string representation from it, and we compare it against the text?

Reply:
This test comes from skara repo. Other than package naming nothing is changed. Should I add a round trip test there itself or in another new test?